homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -358,6 +371,30 @@ def _async_update_volume(self, data: VolumeState) -> None:
 
         self.async_write_ha_state()
 
+    @callback
+    async def _async_update_sound_modes(
```

Comment:
I think `@callback` is not allowed in awaitable functions (`async def`).

Reply:
Yeah ok, should I run the existing method as a Hass job then? For example:
```python
@callback
def _async_update_sound_modes(
    self, active_sound_mode: ListeningModeProps | ListeningModeRef | None = None
) -> None:
    """Update the available sound modes."""
    self.hass.add_job(self._async_update_sound_modes_job, active_sound_mode)
```

It works, I'll just need to update tests.

Reply:
This is still relevant

Reply:
Just remove the decorator. The dispatch helper knows how to call coroutine functions as well as async callbacks.